src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    let workingSeparator = separator;
+
+    if (workingSeparator === null) {
+      workingSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
```

Comment:
Change your `if` statement.
Add `element` to `string` only if `element` !== `null` or !== `undefined` .
Check @naziamoff comments